browser/brave_vpn/dns/brave_vpn_dns_observer_service_win.cc

```diff
@@ -94,7 +95,17 @@ void BraveVpnDnsObserverService::UnlockDNS() {
       ->UpdateNetworkService(false);
 }
 
+bool BraveVpnDnsObserverService::IsDNSHelperLive() {
+  if (dns_helper_live_for_testing_.has_value())
+    return dns_helper_live_for_testing_.value();
+  // If BraveVpnHelper is live we should not override DNS because it will be
+  // handled by the service.
+  return brave_vpn::IsBraveVPNHelperServiceLive();
+}
+
 void BraveVpnDnsObserverService::LockDNS() {
+  if (IsDNSHelperLive())
```

Comment:
We don't need to have this same check at the start of `UnlockDNS()`?

Reply:
no, if we locked and service started later by some reason we have to unlock anyway

Reply:
got it.